spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/amqp/RabbitAutoConfiguration.java

```diff
@@ -93,73 +92,48 @@
 public class RabbitAutoConfiguration {
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnMissingBean(ConnectionFactory.class)
 	protected static class RabbitConnectionFactoryCreator {
 
 		@Bean
-		public CachingConnectionFactory rabbitConnectionFactory(RabbitProperties properties,
+		@ConditionalOnMissingBean
+		RabbitConnectionFactoryBeanConfigurer rabbitConnectionFactoryBeanConfigurer(RabbitProperties properties,
```

Comment:
I know this is not `protected`, but rather `package` visibility. Is there a reason that in general SB will use public methods under protected classes which allow the subclass abuse listed in gh-187 to take place? I am guessing its to allow users to locally bug fix or workaround issues temporarily via the subclass as an immediate fix in odd cases?

Reply:
No, it should be package-private so your change here is a good one.